src/ToricVarieties/NormalToricVarieties/constructors.jl

```diff
@@ -0,0 +1,243 @@
+######################
+# 1: The Julia type for ToricVarieties
+######################
+abstract type AbstractNormalToricVariety end
+
+struct NormalToricVariety <: AbstractNormalToricVariety
+           polymakeNTV::Polymake.BigObject
+end
+export NormalToricVariety
+
+struct AffineNormalToricVariety <: AbstractNormalToricVariety
+           polymakeNTV::Polymake.BigObject
+end
+export AffineNormalToricVariety
+
+
+function pm_ntv(v::AbstractNormalToricVariety)
+    return v.polymakeNTV
+end
+
+######################
+# 2: Generic constructors
+######################
+
+
+@doc Markdown.doc"""
+    AffineNormalToricVariety(C::Cone)
+
+Construct the affine normal toric variety $U_{C}$ corresponding to a polyhedral
+cone `C`.
+
+# Examples
+Set `C` to be the positive orthant in two dimensions.
+```jldoctest
+julia> C = Oscar.positive_hull([1 0; 0 1])
+A polyhedral cone in ambient dimension 2
+
+julia> antv = AffineNormalToricVariety(C)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+```
+"""
+function AffineNormalToricVariety(C::Cone)
+    pmc = Oscar.pm_cone(C)
+    fan = Polymake.fan.check_fan_objects(pmc)
+    pmntv = Polymake.fulton.NormalToricVariety(fan)
+    return AffineNormalToricVariety(pmntv)
+end
+
+
+@doc Markdown.doc"""
+    NormalToricVariety(C::Cone)
+Construct the (affine) normal toric variety $X_{\Sigma}$ corresponding to a
+polyhedral fan $\Sigma = C$ consisting only of the cone `C`.
+# Examples
+Set `C` to be the positive orthant in two dimensions.
+```jldoctest
+julia> C = Oscar.positive_hull([1 0; 0 1])
+A polyhedral cone in ambient dimension 2
+julia> ntv = NormalToricVariety(C)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+```
+"""
+function NormalToricVariety(C::Cone)
```

Comment:
There was a discussion. I wanted `toric_ideal` only to work for an `AffineNormalToricVariety`. In general there are several methods that only work in the affine case. But one can still ask `isaffine(NormalToricVariety)` and it was argued that then `toric_ideal` also should work in this case. But I don't want every function to have the `isaffine` check at the beginning, so then this solution was implemented. Especially since Julia doesn't give us a way for automatic conversion in these cases. I am not religious (anymore) about my arguments.

Nevertheless, if we were to check `isaffine` in every function that needs it, why introduce the `AffineNTV` at all?

Reply:
OK. I have another question, but not the time to ask now. We should merge this and revisit this later (if needed).

Reply:
> There was a discussion. I wanted `toric_ideal` only to work for an `AffineNormalToricVariety`. In general there are several methods that only work in the affine case. But one can still ask `isaffine(NormalToricVariety)` and it was argued that then `toric_ideal` also should work in this case. But I don't want every function to have the `isaffine` check at the beginning, so then this solution was implemented. Especially since Julia doesn't give us a way for automatic conversion in these cases. I am not religious (anymore) about my arguments.
> 
> Nevertheless, if we were to check `isaffine` in every function that needs it, why introduce the `AffineNTV` at all?

It seems to be that all these ramifications only arise from "intelligent" design decisions; see below.  Seems like what you report already proves my case.